tests/test_sklearn_explain_prediction.py

```diff
@@ -150,3 +163,114 @@ def test_explain_linear_regression(boston_train, reg):
+    clf.fit(X, y)
+    res = explain_prediction(
+        clf, X[0], target_names=target_names, feature_names=feature_names)
+    for expl in format_as_all(res, clf):
+        for target in target_names:
+            assert target in expl
+        assert 'BIAS' in expl
+        assert any(f in expl for f in feature_names)
+    check_targets_scores(res)
+
+
+@pytest.mark.parametrize(['clf'], [
+    [DecisionTreeClassifier()],
+    [ExtraTreesClassifier()],
+    [GradientBoostingClassifier(learning_rate=0.075)],
+    [RandomForestClassifier()],
+])
+def test_explain_tree_clf_binary(clf, iris_train_binary):
+    X, y, feature_names = iris_train_binary
+    clf.fit(X, y)
+    all_expls = []
+    for x in X[:5]:
+        res = explain_prediction(clf, x, feature_names=feature_names)
+        text_expl = format_as_text(res, show=fields.WEIGHTS)
+        print(text_expl)
+        assert '<BIAS>' in text_expl
+        check_targets_scores(res)
+        all_expls.append(text_expl)
+    assert any(f in ''.join(all_expls) for f in feature_names)
+
+
+@pytest.mark.parametrize(['reg'], [
+    [DecisionTreeRegressor()],
+    [ExtraTreesRegressor()],
+    [RandomForestRegressor()],
+])
+def test_explain_tree_regressor_multitarget(reg):
+    X, y = make_regression(n_samples=100, n_targets=3, n_features=10,
+                           random_state=42)
+    reg.fit(X, y)
+    res = explain_prediction(reg, X[0])
+    for expl in format_as_all(res, reg):
+        for target in ['y0', 'y1', 'y2']:
+            assert target in expl
+        assert 'BIAS' in expl
+        assert any('x%d' % i in expl for i in range(10))
+    check_targets_scores(res)
+
+
+@pytest.mark.parametrize(['reg'], [
+    [DecisionTreeRegressor()],
+    [ExtraTreesRegressor()],
+    [GradientBoostingRegressor(learning_rate=0.075)],
+    [RandomForestRegressor()],
+])
+def test_explain_tree_regressor(reg, boston_train):
+    X, y, feature_names = boston_train
+    reg.fit(X, y)
+    all_expls = []
+    for i, x in enumerate(X[:5]):
+        res = explain_prediction(reg, x, feature_names=feature_names)
+        text_expl = format_as_text(res, show=fields.WEIGHTS)
+        print(text_expl)
+        assert '<BIAS>' in text_expl
+        check_targets_scores(res)
+        all_expls.append(text_expl)
+    assert any(f in ''.join(all_expls) for f in feature_names)
+
+
+@pytest.mark.parametrize(['clf'], [
+    [DecisionTreeClassifier()],
+    [ExtraTreesClassifier()],
+    [RandomForestClassifier()],
+])
+def test_explain_tree_classifier_text(clf, newsgroups_train_big):
+    docs, y, target_names = newsgroups_train_big
+    vec = CountVectorizer(binary=True, stop_words='english')
+    X = vec.fit_transform(docs)
+    clf.fit(X, y)
+    res = explain_prediction(clf, docs[0], vec=vec, target_names=target_names)
+    check_targets_scores(res)
+    print(format_as_text(res, show=fields.WEIGHTS))
+
+
+def explain_treeinterpreter(clf, xs, feature_names, target_names):
```

Comment:
It's not used, I used it during development to compare with treeinterpreter. Probably we should not integrate it into our test suite, and we have checks that sum of all weights is equal to score.